### PR TITLE
Login

### PR DESCRIPTION
**Pull request by trexdinosaur**
Adds authentication and ability to store models
Also - allows Caffe_App to get migrations... Looks like these got removed accidentally by somebody. 

Longer timeframe (3-4 days):
- Versioning for the models
- User based authentication for a model VS just a URL/ID (think: shareable link in Google Drive VS adding a person...) (@utsavgarg is this something you are looking for? should be very easy to implement.)
- tests for all of this (these might take me awhile... :) )

### CODE REVIEW

File: caffe_app/views/DB.py [hunk 1]
```diff
@@ -25,14 +25,26 @@ def save_to_db(request):
...
+            model.updatedOn = datetime.now()
+            model.name = net_name
             model.save()
-            return JsonResponse({'result': 'success', 'id': randomId})
-        except:
-            return JsonResponse({'result': 'error', 'error': str(sys.exc_info()[1])})
+            return JsonResponse({'result': 'success', 'id': request.POST.get('modelID')})
+        else:
+            try:
+                randomId = datetime.now().strftime('%Y%m%d%H%M%S')+randomword(5)
```
Comment (deshraj) [inline caffe_app/views/DB.py]: Can you please change `randomId` to `random_id` since we don't follow the camel casing in python.

Comment (deshraj) [reply caffe_app/views/DB.py]: Also, can you please make sure that you follow the underscore styling everywhere in the code?

**Comment by trexdinosaur** [comment]
@deshraj request changes are made

File: backendAPI/templates/home.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{% load socialaccount %}
+
+<h1>Django Allauth Tutorial</h1>
+{% if user.is_authenticated %}
+<p>Welcome {{ user.username }} !!!</p>
+{% else %}
+<a href="{% provider_login_url 'github' %}">Sign Up</a>
+{% endif %}
```
[the remark this answers is not in the document]
Comment (trexdinosaur) [reply backendAPI/templates/home.html]: the new line is there, but GitHub is not showing it for some reason....

**Comment by trexdinosaur** [comment]
@deshraj I've made a couple of bug fixes, please check them whenever you have time!

File: backendAPI/views.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from django.views.generic import TemplateView
+from django.http import JsonResponse
+from caffe_app.models import ModelExport
+# from django.shortcuts import render
```
Comment (deshraj) [inline backendAPI/views.py]: Please remove the comments here.

File: backendAPI/views.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# from django.core import serializers
+
+
+# Create your views here.
+class login(TemplateView):
+    template_name = 'home.html'
+
+
+def checkLogin(request):
+    return JsonResponse({'result': request.user.is_authenticated()})
```
Comment (deshraj) [inline backendAPI/views.py]: As mentioned earlier, can you please follow the underscore style instead of camelCasing?

File: backendAPI/views.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+
+def getModels(request):
```
Comment (deshraj) [inline backendAPI/views.py]: Please use underscore styling instead of camelCasing.

File: backendAPI/views.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+from django.views.generic import TemplateView
+from django.http import JsonResponse
+from caffe_app.models import ModelExport
+# from django.shortcuts import render
+# from django.core import serializers
+
+
+# Create your views here.
+class login(TemplateView):
+    template_name = 'home.html'
```
Comment (deshraj) [inline backendAPI/views.py]: Please change the name of the login template from `home.html` to `login.html`

File: backendAPI/views.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+
+def checkLogin(request):
+    return JsonResponse({'result': request.user.is_authenticated()})
+
+
+def getModels(request):
+    if (request.user.is_authenticated()):
+        models = ModelExport.objects.filter(user=request.user).values('name', 'id')
+        #
```
Comment (deshraj) [inline backendAPI/views.py]: Please remove this extra line.

File: .travis.yml [hunk 8]
```diff
@@ -33,6 +33,10 @@ before_script:
   - psql -c "ALTER ROLE admin SET default_transaction_isolation TO 'read committed'" -U postgres
   - psql -c "ALTER ROLE admin SET timezone TO 'UTC'" -U postgres
   - psql -c "ALTER USER admin CREATEDB" -U postgres
+  - python manage.py makemigrations
+  - python manage.py migrate sites
+  - python manage.py migrate auth
+  - python manage.py migrate
```
Comment (deshraj) [inline .travis.yml]: There is no need for line 37 and 38 since line 39 will automatically migrate for both the Django apps `sites` and `apps`

**Comment by trexdinosaur** [comment]
@deshraj made requested changes

**Comment by gautamjajoo** [comment]
@trexdinosaur would you mind resolving the conflicts so that the pull request can be reviewed?

**Comment by pavanagrawal123** [comment]
@gautamjajoo, I will resolve the issues soon, but I'm not sure if this will be merged right now in its current state, so I don't know how fruitful it would be.

**Comment by deshraj** [comment]
@pavanagrawal123: yeah I agree. Please don't work on it yet. We were not able to test it correctly. I will let you know once we are able to test all the features. We can merge it then.

**Comment by Ram81** [comment]
Support for login was added in #363, closing this.

**Pull request closed without merging**